source/state_representation/include/state_representation/StateType.hpp

```diff
@@ -36,4 +36,56 @@ enum class StateType {
 #endif
 };
 
+/**
+ * @brief Convert state type enum to its corresponding name
+ */
+[[maybe_unused]] static std::string get_state_type_name(const StateType& state_type) {
```

Comment:
nitpick: This might be for @eeberhard to tell but would it be more optimized to have a `std::map<StateType, std::string>` or even `std::map<int, std::string>` as we can convert instead of the `switch`?